std/digest/digest.d

```diff
@@ -992,3 +992,15 @@ unittest
     //necessary, result will have the correct length, but buf will still have it's original
     //length
 }
+
+@safe unittest
+{
+    // Test peek & length
+    import std.digest.crc;
+    auto hash = new WrapperDigest!CRC32();
+    assert(hash.length == 4);
+    hash.put(cast(const(ubyte[]))"The quick brown fox jumps over the lazy dog");
+    assert(hash.peek().toHexString() == "39A34F41");
+    ubyte[5] buf;
+    assert(hash.peek(buf).toHexString() == "39A34F41");
```

Comment:
This is supposed to test that it uses the buffer you passed into it, yes? Why not do something like:

```D
assert(hash.peek(buf) is buf);
assert(buf.toHexString == ...);
```

That would ensure that it's filling the input buffer and yielding the correct value, rather than just returning the correct value.